subxt/src/blocks/extrinsic_types.rs

```diff
@@ -558,6 +671,101 @@ impl<T: Config> ExtrinsicEvents<T> {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct ExtrinsicSignedExtensions<'a, T: Config> {
+    signed_extensions: Vec<ExtrinsicSignedExtension<'a, T>>,
+    bytes: &'a [u8],
+    ty_id: u32,
+    metadata: &'a Metadata,
```

Comment:
Prefer `metadata.clone()` I think to avoid the lifetime here (it's an Arc anyway).

Since the bytes are already `Arc<[u8]>` to be cheaply shared, we could avoid all lifetimes and just keep the bytes and start/end indexes as above. I'm torn on this one. It would make it easier for people to store away ExtrinsicSignedExtensions etc though :)

Reply:
Hmm, maybe that is better. But I don't like about it that the `ExtrinsicSignedExtensions` would then contain a pointer to the bytes of the entire extrinsic instead of just having the bytes relating to the actual extra fields of signed extensions. I think this would then require storing start and end indices as well?

Reply:
Yeah; if you had a pointer to all of the bytes you'd need indices as well. I guess it's a choice between user ergonomics (Arc of bytes + indices) versus optimal performance (lifetimes + borrow the bytes we need).

I'm sortof happy either way for now; we could stick to lifetimes and move to Arc+indices if it proves to be annoying for people :)